Common/Product/SharedProject/HierarchyNode.cs

```diff
@@ -1172,28 +1172,40 @@ internal abstract class HierarchyNode :
         }
 
         /// <summary>
+        /// Handles the exclude from project command. After operations 
+        /// is completed it refreshes property browser.
+        /// </summary>
+        internal virtual int ExcludeFromProjectWithRefresh() {
```

Comment:
If ExcludeFromProject is virtual, does this need to be virtual? What other ways might we want to do this?